File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void solve(int i, int j, vector<vector<int>>& ans, int n, int m) {
        for (int col = 0; col < m; col++) {
            ans[i][col] = 0;
        }

        for (int row = 0; row < n; row++) {
            ans[row][j] = 0;
        }
    }

    void setZeroes(vector<vector<int>>& matrix) {
        int n = matrix.size();
        int m = matrix[0].size();
        vector<vector<int>> ans;
        ans = matrix;
        
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (matrix[i][j] == 0) {
                    solve(i, j, ans, n, m);
                }
            }
        }
        
        matrix = ans;
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
            ArrayList<Integer>x = new ArrayList<>();
            ArrayList<Integer>y = new ArrayList<>();
            
            for(int i=0; i<matrix.length; i++){
                    for(int j=0; j<matrix[0].length; j++){
                            if(matrix[i][j] == 0){
                                    x.add(i);
                                    y.add(j);
                            }
                    }
            }
            
//             for rows
            for(int i=0; i<x.size(); i++){
                    int idx = x.get(i);
                    for(int j=0; j<matrix[0].length; j++){
                            matrix[idx][j] = 0; 
                    }
            }
//             for col
            for(int i=0; i<y.size(); i++){
                    int idx = y.get(i);
                    for(int j=0; j<matrix.length; j++){
                            matrix[j][idx] = 0;
                    }
            }
    }
}